__tests__/new/hathorwallet.test.js

```diff
@@ -1003,6 +1063,153 @@ describe('prepare transactions without signature', () => {
       }),
     ]));
   });
+
+  test('prepareMeltTokensData with data outputs', async () => {
+    // fake stuff to support the test
+    const fakeTokenToMeltUtxo = {
+      txId: '002abde4018935e1bbde9600ef79c637adf42385fb1816ec284d702b7bb9ef5f',
+      index: 0,
+      value: 100,
+      token: '01',
+      address: fakeAddress.base58,
+      authorities: 0,
+      timelock: null,
+      locked: false,
+    };
+    const fakeMeltAuthority = [{
+      txId: '002abde4018935e1bbde9600ef79c637adf42385fb1816ec284d702b7bb9ef5f',
+      index: 0,
+      value: 1,
+      token: '01',
+      address: fakeAddress.base58,
+      authorities: TOKEN_MELT_MASK,
+      timelock: null,
+      locked: false,
+    }];
+
+    // wallet and mocks
+    const hWallet = new FakeHathorWallet();
+    hWallet.storage = getStorage({
+      readOnly: false,
+      currentAddress: fakeAddress.base58,
+      selectUtxos: generateSelectUtxos(fakeTokenToMeltUtxo),
+    });
+    jest.spyOn(hWallet, 'getMeltAuthority').mockReturnValue(fakeMeltAuthority);
+
+    // prepare melt
+    const txData = await hWallet.prepareMeltTokensData('01', 100, {
+      address: fakeAddress.base58,
+      unshiftData: true,
+      data: ['foobar'],
+      pinCode: '1234',
+      signTx: false, // skip the signature
+    });
+
+    // assert the transaction is not signed
+    expect(txData.inputs).toHaveLength(2);
+    expect(txData.inputs).toEqual(expect.arrayContaining([
+      expect.objectContaining({
+        data: null,
+      }),
+      expect.objectContaining({
+        data: null,
+      }),
+    ]));
+    // outputs: data + authority
+    expect(txData.outputs).toHaveLength(2);
+    expect(txData.outputs).toEqual(expect.arrayContaining([
+      expect.objectContaining({
+        script: Buffer.from([6, 102, 111, 111, 98, 97, 114, 172]),
+        tokenData: 0,
+        value: 1,
+      }),
+      expect.objectContaining({
+        tokenData: 129,
+        value: 2,
+      }),
+    ]));
+  });
+
+  test('prepareMeltTokensData with data outputs and selecting utxos', async () => {
```

Comment:
This test is very similar to the one above, but we create 2 data outputs, so we need to find 0,01 HTR to add on the inputs, the utxo found has 0.02 HTR so we need to add the change on the outputs.
So we add 0.02 HTR on the inputs and 2 new outputs (data + change)